### PR TITLE
Refactoring: adds _ID const to all the library elements

### PR DESCRIPTION
**Pull request by MarcinOrlowski**
This PR is similar to #648 and adds `_ID` constant to all the tools and refactors the code to use it instead of hard-coded string references.

### CODE REVIEW

File: src/main/java/com/cburch/gray/GrayCounter.java [hunk 1]
```diff
@@ -48,8 +48,16 @@
  * features beyond the SimpleGrayCounter class.
  */
 class GrayCounter extends InstanceFactory {
+  /**
+   * Unique identifier of the tool, used as reference in project files.
+   * Do NOT change as it will prevent project files from loading.
+   *
+   * Identifier value must MUST be unique string among all tools.
+   */
+  public static final String _ID = "Gray Counter";
```
Comment (maehne) [inline src/main/java/com/cburch/gray/GrayCounter.java:57]: What is your reasoning for adding the underscore in front of `ID`? Other member variables don't use such a prefix. Are you fearing it is to short and could be easily shadowed or shadow something else?

File: src/main/java/com/cburch/logisim/file/XmlReader.java [hunk 2]
```diff
@@ -1195,11 +1195,13 @@ private void repairForWiringLibrary(Document doc, Element root) {
         labelMap,
         oldBaseLabel,
         newBaseLabel,
+        // FIXME: we should use _IDs here, right?
```
Comment (maehne) [inline src/main/java/com/cburch/logisim/file/XmlReader.java:1198]: This seems sensible to me. @BFH-ktt1: What do you think?

File: src/main/java/com/cburch/logisim/file/XmlReader.java [hunk 3]
```diff
@@ -1195,11 +1195,13 @@ private void repairForWiringLibrary(Document doc, Element root) {
...
         "Poke Tool;" + "Edit Tool;Select Tool;Wiring Tool;Text Tool;Menu Tool;Text");
     addToLabelMap(
         labelMap,
         oldBaseLabel,
         wiringLabel,
+        // FIXME: we should use _IDs here, right?
```
Comment (maehne) [inline src/main/java/com/cburch/logisim/file/XmlReader.java:1204]: This seems sensible to me. @BFH-ktt1: What do you think?

**Review by maehne — APPROVED** [review]
Looks good to me! Thanks for the refactoring @MarcinOrlowski!

Due to its size, I'll request an additional review from @BFH-ktt1.

File: src/main/java/com/cburch/gray/GrayCounter.java (hunk 1 above)
Comment (MarcinOrlowski) [reply src/main/java/com/cburch/gray/GrayCounter.java:57]: Right now, for consistency a with #648 as `_ID` landed there already. I do not think there can be a conflict of any sort and I can have all the `_ID` replaced with just `ID` in next PR.

**Comment by BFH-ktt1** [comment]
@MarcinOrlowski : Thanks for the work, could you just look into the conflicts, such that I can merge?

File: src/main/java/com/cburch/logisim/file/XmlReader.java (hunk 3 above)
Comment (MarcinOrlowski) [reply src/main/java/com/cburch/logisim/file/XmlReader.java:1201]: I think we do, but this also means that all other tools (i.e. "Poke Tool" etc) must be also amended to have its own `_ID` so all these strings here can be replaced by use of consts. I fix that later.

**Comment by MarcinOrlowski** [comment]
This one is up to date now and can be merged.

**Pull request merged**